cypress/e2e/challenges.cy.js

```diff
@@ -88,4 +89,13 @@ describe('Challenge Tests', () => {
     cy.dataCy(ChallengesPage.SUCCESS_ALERT).should('contain', 'Your answer is correct!')
     cy.dataCy(ChallengesPage.PROGRESS_BAR).should('be.visible').should('not.have.attr', 'aria-valuenow', '0')
   })
+
+  it('Submitting right answer gives visual cue on homepage that the challenge is successfully solved', () => {
+    cy.visit('/challenge/0')
+    cy.dataCy(ChallengesPage.ANSWER_TEXTBOX).type('The first answer')
+    cy.dataCy(ChallengesPage.SUBMIT_TEXTBOX_BTN).click()
+    cy.dataCy(ChallengesPage.SUCCESS_ALERT).should('contain', 'Your answer is correct!')
+    cy.visit('/')
+    cy.dataCy(HomePage.CHALLENGE_TABLE_ROW).first().should('have.class', 'solved')
```

Comment:
I think we may be able to label each of these rows like was done with the challenge links. This is how it was done on the challenge link 
```
<td>
                                        &nbsp;<span
                                            class="d-xl-none"
                                            th:if="${completedChallenges != null && #lists.contains(completedChallenges, '' + challenge.link)}"
                                            >&#9745; </span
                                        ><a
                                            th:href="'/challenge/' + ${challenge.link}"
                                            th:class="${challenge.isChallengeEnabled} ? '' : 'disabled'"
                                            ><span
                                                th:text="${challenge.name}"
                                                th:attr="data-cy=${#strings.toLowerCase(challenge.name)} + '-link'"
                                            ></span>
                                        </a>
                                    </td>
```

It would just be a bit more explicit than .first() which I think could potentially cause problems

Reply:
Maybe we can have labels for disabled links as well then? That way we don't have the issues I tried to fix in https://github.com/OWASP/wrongsecrets/pull/804/commits/d7e1a1c27e007c4b7a1149f295110eac82afe4dd and then we can test for those disabled labels being present or not...

Reply:
Labels have been added, now all that is needed, might be a label-based ui test instead :)